src/libraries/System.Private.CoreLib/src/System/Collections/Generic/INonRandomizedEqualityComparer.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Collections.Generic
+{
+    internal interface INonRandomizedEqualityComparer<in T> : IEqualityComparer<T>
```

Comment:
Nit: Regardless of whether we keep this interface in the end, I'd say I find the name a bit confusing, in the sense that most IEqualityComparer instances out there are non-randomized but do not implement this subtype. Perhaps IRandomizableEqualityComparer might be more suitable?

Reply:
Maybe `IFastEqualityComparer<in T>`? The premise of the interface is "My `GetHashCode` routine is faster than a normal `GetHashCode` routine, but I'm not collision-resistant. I'm intended as an implementation detail for collections who want optimistically to assume the data isn't malicious, but I also provide a fallback in case things go bad."